crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -30,7 +33,14 @@ pub struct SsaContext {
     value_names: HashMap<NodeId, u32>,
     pub sealed_blocks: HashSet<BlockId>,
     pub mem: Memory,
+
     pub functions: HashMap<FuncId, function::SSAFunction>,
+    pub function_ids: HashMap<FuncId, NodeId>,
+    pub opcode_ids: HashMap<OPCODE, NodeId>,
+
+    /// Maps ArrayIdSet -> Vec<(ArrayId, u32)>,
+    pub function_returned_arrays: Vec<Vec<(ArrayId, /*result_index:*/ u32)>>,
```

Comment:
The handling of returned arrays is wrong, they should stay with the call instruction